RHIDP-1471: add GitHub auth provider content (#91)
redhat-developer/red-hat-developers-documentation-rhdh: getting-started/proc-configuring-github-app.adoc

[id="proc-configuring-github-app"]

> = Configuring a GitHub App

Reviewer: We elaborate on the heading for the user's clarity. Where the user needs to configure the GitHub app?

The GitHub authentication provider uses the following configuration keys:

> * `clientId`: the client ID that you generated on GitHub, e.g. b59241722e3c3b4816e2

Reviewer: ```suggestion
* `clientId`: the client ID that you generated on GitHub, such as `b59241722e3c3b4816e2`
```

* `clientSecret`: the client secret tied to the generated client ID.

> * `enterpriseInstanceUrl` (optional): the base URL for a GitHub Enterprise instance, e.g. https://ghe.<company>.com. Only needed for GitHub Enterprise.

Reviewer: ```suggestion
* `enterpriseInstanceUrl` (optional): the base URL for a GitHub Enterprise instance, such as `https://ghe.<company>.com`. Only needed for GitHub enterprise.
```

> * `callbackUrl` (optional): the callback URL that GitHub will use when initiating an OAuth flow, e.g. https://your-intermediate-service.com/handler. Only needed if {product-short} is not the immediate receiver (e.g. one OAuth app for many {product-short} instances).

Reviewer: ```suggestion
* `callbackUrl` (optional): the callback URL that GitHub uses when initiating an OAuth flow, such as `https://your-intermediate-service.com/handler`. Only needed if {product-short} is not the immediate receiver (For example one OAuth app for many {product-short} instances).
```
